cli/commands/watch/deployer.ts

```
@@ -0,0 +1,155 @@
+		catch (error) {
+			this.status = 'dfx-error';
+			onProgress();
+			return;
+		}
+
+		this.status = 'running';
+		onProgress();
+
+		// create canisters (sequentially to avoid DFX errors)
+		let resolve : (() => void) | undefined;
+		this.currentRun = new Promise<void>((res) => resolve = res);
```

Biome

[error] The assignment should not be in an expression.

The use of assignments in expressions is confusing.
Expressions are often considered as side-effect free.

(lint/suspicious/noAssignInExpressions)